java/config/checkstyle/checkstyle.xml

```diff
@@ -26,56 +26,64 @@
 
 <module name="Checker">
 
+  <!-- General configuration for Checkstyle -->
+  <!-- See https://checkstyle.org/config.html#Checker -->
   <property name="charset" value="UTF-8"/>
-
+  <property name="tabWidth" value="4"/>
+  <property name="fileExtensions" value="java"/>
   <property name="severity" value="${org.checkstyle.google.severity}" default="warning"/>
 
-  <!-- It's standard practice to separate any exceptions/suppressions into a separate XML file. -->
-  <module name="SuppressionFilter">
-    <property name="file" value="${config_loc}/suppressions.xml" />
-  </module>
-
-  <property name="fileExtensions" value="java, properties, xml"/>
-  <!-- Excludes all 'module-info.java' files              -->
+  <!-- Fine-grained exclusion rules -->
   <!-- See https://checkstyle.org/filefilters/index.html -->
   <module name="BeforeExecutionExclusionFileFilter">
+    <!-- Exclude all 'module-info.java' files -->
     <property name="fileNamePattern" value="module\-info\.java$"/>
   </module>
 
-  <module name="SuppressWarningsFilter"/>
```

Comment:
I removed this rule, it allows you to suppress checkstyle warnings with a Java annotation.
We always use comments (more flexible) to toggle checkstyle, not annotations.